commit 87ab6c205f36b98401e92980dc32910e40ab83e1
Author: Rubaka <blackmanos@yandex.ru>
Date:   Thu Nov 7 13:52:48 2013 +0200

    РїРѕРїСЂРѕР±СѓРµРј РІС‹РґРµР»РёС‚СЊ РІ Р»РёР±Сѓ

diff --git a/src/server/game/CMakeLists.txt b/src/server/game/CMakeLists.txt
index b74eae0..26eb8c2 100644
--- a/src/server/game/CMakeLists.txt
+++ b/src/server/game/CMakeLists.txt
@@ -243,6 +243,7 @@ include_directories(
   ${CMAKE_CURRENT_SOURCE_DIR}/Weather
   ${CMAKE_CURRENT_SOURCE_DIR}/World
   ${CMAKE_SOURCE_DIR}/src/server/scripts/PrecompiledHeaders
+  ${CMAKE_SOURCE_DIR}/src/server/scripts/EasternKingdoms/PrecompiledHeaders
   ${ACE_INCLUDE_DIR}
   ${MYSQL_INCLUDE_DIR}
   ${OPENSSL_INCLUDE_DIR}
diff --git a/src/server/scripts/CMakeLists.txt b/src/server/scripts/CMakeLists.txt
index 0213a63..db2757d 100644
--- a/src/server/scripts/CMakeLists.txt
+++ b/src/server/scripts/CMakeLists.txt
@@ -40,7 +40,8 @@ if(SCRIPTS)
   include(Custom/CMakeLists.txt)
   include(World/CMakeLists.txt)
   include(OutdoorPvP/CMakeLists.txt)
-  include(EasternKingdoms/CMakeLists.txt)
+  #include(EasternKingdoms/CMakeLists.txt)
+  add_subdirectory(EasternKingdoms)
   include(Kalimdor/CMakeLists.txt)
   include(Outland/CMakeLists.txt)
   include(Northrend/CMakeLists.txt)
diff --git a/src/server/scripts/EasternKingdoms/BaradinHold/boss_alizabal.cpp b/src/server/scripts/EasternKingdoms/BaradinHold/boss_alizabal.cpp
index 76f6e5a..b0a5b23 100644
--- a/src/server/scripts/EasternKingdoms/BaradinHold/boss_alizabal.cpp
+++ b/src/server/scripts/EasternKingdoms/BaradinHold/boss_alizabal.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "baradin_hold.h"
 
 enum ScriptTexts
diff --git a/src/server/scripts/EasternKingdoms/BaradinHold/boss_occuthar.cpp b/src/server/scripts/EasternKingdoms/BaradinHold/boss_occuthar.cpp
index 5b70d7f..05d77a6 100644
--- a/src/server/scripts/EasternKingdoms/BaradinHold/boss_occuthar.cpp
+++ b/src/server/scripts/EasternKingdoms/BaradinHold/boss_occuthar.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "baradin_hold.h"
 
 enum Spells
diff --git a/src/server/scripts/EasternKingdoms/BaradinHold/instance_baradin_hold.cpp b/src/server/scripts/EasternKingdoms/BaradinHold/instance_baradin_hold.cpp
index 976c9dd..50f5e97 100644
--- a/src/server/scripts/EasternKingdoms/BaradinHold/instance_baradin_hold.cpp
+++ b/src/server/scripts/EasternKingdoms/BaradinHold/instance_baradin_hold.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "baradin_hold.h"
 
 DoorData const doorData[] =
diff --git a/src/server/scripts/EasternKingdoms/BastionOfTwilight/bastion_of_twilight.cpp b/src/server/scripts/EasternKingdoms/BastionOfTwilight/bastion_of_twilight.cpp
index 5116d6d..a78c30a 100644
--- a/src/server/scripts/EasternKingdoms/BastionOfTwilight/bastion_of_twilight.cpp
+++ b/src/server/scripts/EasternKingdoms/BastionOfTwilight/bastion_of_twilight.cpp
@@ -1,4 +1,4 @@
-#include"ScriptPCH.h"
+#include"EasternKingdomsPCH.h"
 #include "bastion_of_twilight.h"
 
 enum ScriptTexts
diff --git a/src/server/scripts/EasternKingdoms/BastionOfTwilight/boss_chogall.cpp b/src/server/scripts/EasternKingdoms/BastionOfTwilight/boss_chogall.cpp
index 45a09d5..3f1307e 100644
--- a/src/server/scripts/EasternKingdoms/BastionOfTwilight/boss_chogall.cpp
+++ b/src/server/scripts/EasternKingdoms/BastionOfTwilight/boss_chogall.cpp
@@ -1,4 +1,4 @@
-#include"ScriptPCH.h"
+#include"EasternKingdomsPCH.h"
 #include"Spell.h"
 #include"bastion_of_twilight.h"
 
diff --git a/src/server/scripts/EasternKingdoms/BastionOfTwilight/boss_council_of_ascendents.cpp b/src/server/scripts/EasternKingdoms/BastionOfTwilight/boss_council_of_ascendents.cpp
index d0ed763..4a18cef 100644
--- a/src/server/scripts/EasternKingdoms/BastionOfTwilight/boss_council_of_ascendents.cpp
+++ b/src/server/scripts/EasternKingdoms/BastionOfTwilight/boss_council_of_ascendents.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "Spell.h"
 #include "bastion_of_twilight.h"
 
diff --git a/src/server/scripts/EasternKingdoms/BastionOfTwilight/boss_halfus_wyrmbreaker.cpp b/src/server/scripts/EasternKingdoms/BastionOfTwilight/boss_halfus_wyrmbreaker.cpp
index 9f94041..33b2c41 100644
--- a/src/server/scripts/EasternKingdoms/BastionOfTwilight/boss_halfus_wyrmbreaker.cpp
+++ b/src/server/scripts/EasternKingdoms/BastionOfTwilight/boss_halfus_wyrmbreaker.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "Spell.h"
 #include "bastion_of_twilight.h"
 
diff --git a/src/server/scripts/EasternKingdoms/BastionOfTwilight/boss_sinestra.cpp b/src/server/scripts/EasternKingdoms/BastionOfTwilight/boss_sinestra.cpp
index 9bd17c0..f18797e 100644
--- a/src/server/scripts/EasternKingdoms/BastionOfTwilight/boss_sinestra.cpp
+++ b/src/server/scripts/EasternKingdoms/BastionOfTwilight/boss_sinestra.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "Spell.h"
 #include "bastion_of_twilight.h"
 
diff --git a/src/server/scripts/EasternKingdoms/BastionOfTwilight/boss_theralion_and_valiona.cpp b/src/server/scripts/EasternKingdoms/BastionOfTwilight/boss_theralion_and_valiona.cpp
index dc1bcc2..2e68eed 100644
--- a/src/server/scripts/EasternKingdoms/BastionOfTwilight/boss_theralion_and_valiona.cpp
+++ b/src/server/scripts/EasternKingdoms/BastionOfTwilight/boss_theralion_and_valiona.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "bastion_of_twilight.h"
 
 enum ValionaScriptTexts
diff --git a/src/server/scripts/EasternKingdoms/BastionOfTwilight/instance_bastion_of_twilight.cpp b/src/server/scripts/EasternKingdoms/BastionOfTwilight/instance_bastion_of_twilight.cpp
index 79b1c04..895b6d9 100644
--- a/src/server/scripts/EasternKingdoms/BastionOfTwilight/instance_bastion_of_twilight.cpp
+++ b/src/server/scripts/EasternKingdoms/BastionOfTwilight/instance_bastion_of_twilight.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "bastion_of_twilight.h"
 
 #define MAX_ENCOUNTER 5
diff --git a/src/server/scripts/EasternKingdoms/BlackrockCaverns/blackrock_caverns_teleport.cpp b/src/server/scripts/EasternKingdoms/BlackrockCaverns/blackrock_caverns_teleport.cpp
index 3411f57..210062a 100644
--- a/src/server/scripts/EasternKingdoms/BlackrockCaverns/blackrock_caverns_teleport.cpp
+++ b/src/server/scripts/EasternKingdoms/BlackrockCaverns/blackrock_caverns_teleport.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "blackrock_caverns.h"
 
 class blackrock_caverns_teleport: public CreatureScript
diff --git a/src/server/scripts/EasternKingdoms/BlackrockCaverns/boss_ascendant_lord_obsidius.cpp b/src/server/scripts/EasternKingdoms/BlackrockCaverns/boss_ascendant_lord_obsidius.cpp
index 286780d..2fb6b33 100644
--- a/src/server/scripts/EasternKingdoms/BlackrockCaverns/boss_ascendant_lord_obsidius.cpp
+++ b/src/server/scripts/EasternKingdoms/BlackrockCaverns/boss_ascendant_lord_obsidius.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "blackrock_caverns.h"
 
 //todo: сделать спелл трансформации
diff --git a/src/server/scripts/EasternKingdoms/BlackrockCaverns/boss_beauty.cpp b/src/server/scripts/EasternKingdoms/BlackrockCaverns/boss_beauty.cpp
index b4cae8c..db8f397 100644
--- a/src/server/scripts/EasternKingdoms/BlackrockCaverns/boss_beauty.cpp
+++ b/src/server/scripts/EasternKingdoms/BlackrockCaverns/boss_beauty.cpp
@@ -1,5 +1,5 @@
 
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "blackrock_caverns.h"
 
 //todo: реализовать касты щенков, берсерк при убийстве ранти, взрыв magma split
diff --git a/src/server/scripts/EasternKingdoms/BlackrockCaverns/boss_corla_hearld_of_twilight.cpp b/src/server/scripts/EasternKingdoms/BlackrockCaverns/boss_corla_hearld_of_twilight.cpp
index bcf4bba..463e67e 100644
--- a/src/server/scripts/EasternKingdoms/BlackrockCaverns/boss_corla_hearld_of_twilight.cpp
+++ b/src/server/scripts/EasternKingdoms/BlackrockCaverns/boss_corla_hearld_of_twilight.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "Spell.h"
 #include "blackrock_caverns.h"
 
diff --git a/src/server/scripts/EasternKingdoms/BlackrockCaverns/boss_karsh_steelbender.cpp b/src/server/scripts/EasternKingdoms/BlackrockCaverns/boss_karsh_steelbender.cpp
index 26ed745..f989d64 100644
--- a/src/server/scripts/EasternKingdoms/BlackrockCaverns/boss_karsh_steelbender.cpp
+++ b/src/server/scripts/EasternKingdoms/BlackrockCaverns/boss_karsh_steelbender.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "blackrock_caverns.h"
 
 //todo: сделать скалирование дамага от стаков брони
diff --git a/src/server/scripts/EasternKingdoms/BlackrockCaverns/boss_romogg_bonecrusher.cpp b/src/server/scripts/EasternKingdoms/BlackrockCaverns/boss_romogg_bonecrusher.cpp
index 9af934f..55dd2d1 100644
--- a/src/server/scripts/EasternKingdoms/BlackrockCaverns/boss_romogg_bonecrusher.cpp
+++ b/src/server/scripts/EasternKingdoms/BlackrockCaverns/boss_romogg_bonecrusher.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "blackrock_caverns.h"
 
 //todo: реализовать призыв о помощи
diff --git a/src/server/scripts/EasternKingdoms/BlackrockCaverns/instance_blackrock_caverns.cpp b/src/server/scripts/EasternKingdoms/BlackrockCaverns/instance_blackrock_caverns.cpp
index a136b0c..6f09ca2 100644
--- a/src/server/scripts/EasternKingdoms/BlackrockCaverns/instance_blackrock_caverns.cpp
+++ b/src/server/scripts/EasternKingdoms/BlackrockCaverns/instance_blackrock_caverns.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "blackrock_caverns.h"
 
 #define ENCOUNTERS 5
diff --git a/src/server/scripts/EasternKingdoms/BlackwingDescent/blackwing_descent.cpp b/src/server/scripts/EasternKingdoms/BlackwingDescent/blackwing_descent.cpp
index 1381b34..09184dd 100644
--- a/src/server/scripts/EasternKingdoms/BlackwingDescent/blackwing_descent.cpp
+++ b/src/server/scripts/EasternKingdoms/BlackwingDescent/blackwing_descent.cpp
@@ -1,4 +1,4 @@
-#include"ScriptPCH.h"
+#include"EasternKingdomsPCH.h"
 #include "blackwing_descent.h"
 
 enum ScriptTexts
diff --git a/src/server/scripts/EasternKingdoms/BlackwingDescent/boss_atramedes.cpp b/src/server/scripts/EasternKingdoms/BlackwingDescent/boss_atramedes.cpp
index 18bf1da..d44d29f 100644
--- a/src/server/scripts/EasternKingdoms/BlackwingDescent/boss_atramedes.cpp
+++ b/src/server/scripts/EasternKingdoms/BlackwingDescent/boss_atramedes.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "Spell.h"
 #include "blackwing_descent.h"
 
diff --git a/src/server/scripts/EasternKingdoms/BlackwingDescent/boss_chimaeron.cpp b/src/server/scripts/EasternKingdoms/BlackwingDescent/boss_chimaeron.cpp
index 51411cf..8e65363 100644
--- a/src/server/scripts/EasternKingdoms/BlackwingDescent/boss_chimaeron.cpp
+++ b/src/server/scripts/EasternKingdoms/BlackwingDescent/boss_chimaeron.cpp
@@ -1,4 +1,4 @@
-#include"ScriptPCH.h"
+#include"EasternKingdomsPCH.h"
 #include"Spell.h"
 #include"blackwing_descent.h"
 
diff --git a/src/server/scripts/EasternKingdoms/BlackwingDescent/boss_magmaw.cpp b/src/server/scripts/EasternKingdoms/BlackwingDescent/boss_magmaw.cpp
index 2df24d4..aedbca7 100644
--- a/src/server/scripts/EasternKingdoms/BlackwingDescent/boss_magmaw.cpp
+++ b/src/server/scripts/EasternKingdoms/BlackwingDescent/boss_magmaw.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "Spell.h"
 #include "blackwing_descent.h"
 
diff --git a/src/server/scripts/EasternKingdoms/BlackwingDescent/boss_maloriak.cpp b/src/server/scripts/EasternKingdoms/BlackwingDescent/boss_maloriak.cpp
index 15291db..d9c20f6 100644
--- a/src/server/scripts/EasternKingdoms/BlackwingDescent/boss_maloriak.cpp
+++ b/src/server/scripts/EasternKingdoms/BlackwingDescent/boss_maloriak.cpp
@@ -1,4 +1,4 @@
-#include"ScriptPCH.h"
+#include"EasternKingdomsPCH.h"
 #include"Spell.h"
 #include "GameObjectAI.h"
 #include "blackwing_descent.h"
diff --git a/src/server/scripts/EasternKingdoms/BlackwingDescent/boss_omnotron_defence_system.cpp b/src/server/scripts/EasternKingdoms/BlackwingDescent/boss_omnotron_defence_system.cpp
index 7957ed5..67ef849 100644
--- a/src/server/scripts/EasternKingdoms/BlackwingDescent/boss_omnotron_defence_system.cpp
+++ b/src/server/scripts/EasternKingdoms/BlackwingDescent/boss_omnotron_defence_system.cpp
@@ -1,4 +1,4 @@
-#include"ScriptPCH.h"
+#include"EasternKingdomsPCH.h"
 #include"Spell.h"
 #include "blackwing_descent.h"
 
diff --git a/src/server/scripts/EasternKingdoms/BlackwingDescent/instance_blackwing_descent.cpp b/src/server/scripts/EasternKingdoms/BlackwingDescent/instance_blackwing_descent.cpp
index b898f33..df08cf5 100644
--- a/src/server/scripts/EasternKingdoms/BlackwingDescent/instance_blackwing_descent.cpp
+++ b/src/server/scripts/EasternKingdoms/BlackwingDescent/instance_blackwing_descent.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "blackwing_descent.h"
 
 #define MAX_ENCOUNTER 6
diff --git a/src/server/scripts/EasternKingdoms/CMakeLists.txt b/src/server/scripts/EasternKingdoms/CMakeLists.txt
index 8e0d381..39de664 100644
--- a/src/server/scripts/EasternKingdoms/CMakeLists.txt
+++ b/src/server/scripts/EasternKingdoms/CMakeLists.txt
@@ -7,259 +7,385 @@
 # This program is distributed in the hope that it will be useful, but
 # WITHOUT ANY WARRANTY, to the extent permitted by law; without even the
 # implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+if( USE_EasternKingdomsPCH )
+  include_directories(
+    ${CMAKE_CURRENT_BINARY_DIR}
+  )
+endif()
 
-set(scripts_STAT_SRCS
-  ${scripts_STAT_SRCS}
-  EasternKingdoms/ghostlands.cpp
-  EasternKingdoms/eversong_woods.cpp
-  EasternKingdoms/AlteracValley/boss_galvangar.cpp
-  EasternKingdoms/AlteracValley/boss_balinda.cpp
-  EasternKingdoms/AlteracValley/boss_drekthar.cpp
-  EasternKingdoms/AlteracValley/boss_vanndar.cpp
-  EasternKingdoms/AlteracValley/alterac_valley.cpp
-  EasternKingdoms/Scholomance/boss_the_ravenian.cpp
-  EasternKingdoms/Scholomance/boss_instructor_malicia.cpp
-  EasternKingdoms/Scholomance/boss_death_knight_darkreaver.cpp
-  EasternKingdoms/Scholomance/boss_illucia_barov.cpp
-  EasternKingdoms/Scholomance/scholomance.h
-  EasternKingdoms/Scholomance/boss_vectus.cpp
-  EasternKingdoms/Scholomance/boss_jandice_barov.cpp
-  EasternKingdoms/Scholomance/boss_kormok.cpp
-  EasternKingdoms/Scholomance/boss_lord_alexei_barov.cpp
-  EasternKingdoms/Scholomance/boss_doctor_theolen_krastinov.cpp
-  EasternKingdoms/Scholomance/boss_darkmaster_gandling.cpp
-  EasternKingdoms/Scholomance/instance_scholomance.cpp
-  EasternKingdoms/Scholomance/boss_lorekeeper_polkelt.cpp
-  EasternKingdoms/Scholomance/boss_ras_frostwhisper.cpp
-  EasternKingdoms/isle_of_queldanas.cpp
-  EasternKingdoms/ZulGurub/boss_grilek.cpp
-  EasternKingdoms/ZulGurub/boss_hazzarah.cpp
-  EasternKingdoms/ZulGurub/boss_jindo_the_godbreaker.cpp
-  EasternKingdoms/ZulGurub/boss_kilnara.cpp
-  EasternKingdoms/ZulGurub/boss_mandokir.cpp
-  EasternKingdoms/ZulGurub/boss_renataki.cpp
-  EasternKingdoms/ZulGurub/boss_venoxis.cpp
-  EasternKingdoms/ZulGurub/boss_wushoolay.cpp
-  EasternKingdoms/ZulGurub/boss_zanzil.cpp
-  EasternKingdoms/ZulGurub/instance_zulgurub.cpp
-  EasternKingdoms/ZulGurub/zulgurub.cpp
-  EasternKingdoms/ZulGurub/zulgurub.h
-  EasternKingdoms/wetlands.cpp
-  EasternKingdoms/arathi_highlands.cpp
-  EasternKingdoms/Gnomeregan/instance_gnomeregan.cpp
-  EasternKingdoms/Gnomeregan/gnomeregan.cpp
-  EasternKingdoms/Gnomeregan/gnomeregan.h
-  EasternKingdoms/redridge_mountains.cpp
-  EasternKingdoms/BlackrockDepths/boss_high_interrogator_gerstahn.cpp
-  EasternKingdoms/BlackrockDepths/boss_gorosh_the_dervish.cpp
-  EasternKingdoms/BlackrockDepths/blackrock_depths.cpp
-  EasternKingdoms/BlackrockDepths/boss_anubshiah.cpp
-  EasternKingdoms/BlackrockDepths/boss_tomb_of_seven.cpp
-  EasternKingdoms/BlackrockDepths/boss_general_angerforge.cpp
-  EasternKingdoms/BlackrockDepths/boss_grizzle.cpp
-  EasternKingdoms/BlackrockDepths/boss_ambassador_flamelash.cpp
-  EasternKingdoms/BlackrockDepths/instance_blackrock_depths.cpp
-  EasternKingdoms/BlackrockDepths/boss_moira_bronzebeard.cpp
-  EasternKingdoms/BlackrockDepths/blackrock_depths.h
-  EasternKingdoms/BlackrockDepths/boss_emperor_dagran_thaurissan.cpp
-  EasternKingdoms/BlackrockDepths/boss_magmus.cpp
-  EasternKingdoms/ironforge.cpp
-  EasternKingdoms/ScarletEnclave/chapter2.cpp
-  EasternKingdoms/ScarletEnclave/chapter5.cpp
-  EasternKingdoms/ScarletEnclave/chapter1.cpp
-  EasternKingdoms/ScarletEnclave/the_scarlet_enclave.cpp
-  EasternKingdoms/eastern_plaguelands.cpp
-  EasternKingdoms/MoltenCore/boss_gehennas.cpp
-  EasternKingdoms/MoltenCore/boss_lucifron.cpp
-  EasternKingdoms/MoltenCore/boss_golemagg.cpp
-  EasternKingdoms/MoltenCore/boss_majordomo_executus.cpp
-  EasternKingdoms/MoltenCore/boss_baron_geddon.cpp
-  EasternKingdoms/MoltenCore/boss_ragnaros.cpp
-  EasternKingdoms/MoltenCore/boss_garr.cpp
-  EasternKingdoms/MoltenCore/molten_core.h
-  EasternKingdoms/MoltenCore/instance_molten_core.cpp
-  EasternKingdoms/MoltenCore/boss_sulfuron_harbinger.cpp
-  EasternKingdoms/MoltenCore/boss_magmadar.cpp
-  EasternKingdoms/MoltenCore/boss_shazzrah.cpp
-  EasternKingdoms/Stratholme/boss_baroness_anastari.cpp
-  EasternKingdoms/Stratholme/boss_nerubenkan.cpp
-  EasternKingdoms/Stratholme/instance_stratholme.cpp
-  EasternKingdoms/Stratholme/boss_dathrohan_balnazzar.cpp
-  EasternKingdoms/Stratholme/boss_timmy_the_cruel.cpp
-  EasternKingdoms/Stratholme/boss_baron_rivendare.cpp
-  EasternKingdoms/Stratholme/boss_magistrate_barthilas.cpp
-  EasternKingdoms/Stratholme/boss_order_of_silver_hand.cpp
-  EasternKingdoms/Stratholme/boss_ramstein_the_gorger.cpp
-  EasternKingdoms/Stratholme/boss_cannon_master_willey.cpp
-  EasternKingdoms/Stratholme/boss_maleki_the_pallid.cpp
-  EasternKingdoms/Stratholme/boss_postmaster_malown.cpp
-  EasternKingdoms/Stratholme/stratholme.h
-  EasternKingdoms/Stratholme/stratholme.cpp
-  EasternKingdoms/tirisfal_glades.cpp
-  EasternKingdoms/SunkenTemple/sunken_temple.cpp
-  EasternKingdoms/SunkenTemple/sunken_temple.h
-  EasternKingdoms/SunkenTemple/instance_sunken_temple.cpp
-  EasternKingdoms/MagistersTerrace/boss_felblood_kaelthas.cpp
-  EasternKingdoms/MagistersTerrace/magisters_terrace.h
-  EasternKingdoms/MagistersTerrace/boss_priestess_delrissa.cpp
-  EasternKingdoms/MagistersTerrace/instance_magisters_terrace.cpp
-  EasternKingdoms/MagistersTerrace/boss_selin_fireheart.cpp
-  EasternKingdoms/MagistersTerrace/boss_vexallus.cpp
-  EasternKingdoms/MagistersTerrace/magisters_terrace.cpp
-  EasternKingdoms/Uldaman/uldaman.cpp
-  EasternKingdoms/Uldaman/boss_ironaya.cpp
-  EasternKingdoms/Uldaman/uldaman.h
-  EasternKingdoms/Uldaman/instance_uldaman.cpp
-  EasternKingdoms/Uldaman/boss_archaedas.cpp
-  EasternKingdoms/swamp_of_sorrows.cpp
-  EasternKingdoms/BlackrockSpire/boss_pyroguard_emberseer.cpp
-  EasternKingdoms/BlackrockSpire/boss_drakkisath.cpp
-  EasternKingdoms/BlackrockSpire/boss_warmaster_voone.cpp
-  EasternKingdoms/BlackrockSpire/boss_mother_smolderweb.cpp
-  EasternKingdoms/BlackrockSpire/boss_quartermaster_zigris.cpp
-  EasternKingdoms/BlackrockSpire/boss_halycon.cpp
-  EasternKingdoms/BlackrockSpire/boss_overlord_wyrmthalak.cpp
-  EasternKingdoms/BlackrockSpire/boss_shadow_hunter_voshgajin.cpp
-  EasternKingdoms/BlackrockSpire/boss_gyth.cpp
-  EasternKingdoms/BlackrockSpire/boss_rend_blackhand.cpp
-  EasternKingdoms/BlackrockSpire/boss_highlord_omokk.cpp
-  EasternKingdoms/BlackrockSpire/boss_the_beast.cpp
-  EasternKingdoms/BlackrockSpire/blackrock_spire.h
-  EasternKingdoms/BlackrockSpire/instance_blackrock_spire.cpp
-  EasternKingdoms/SunwellPlateau/boss_eredar_twins.cpp
-  EasternKingdoms/SunwellPlateau/boss_kiljaeden.cpp
-  EasternKingdoms/SunwellPlateau/sunwell_plateau.h
-  EasternKingdoms/SunwellPlateau/boss_muru.cpp
-  EasternKingdoms/SunwellPlateau/instance_sunwell_plateau.cpp
-  EasternKingdoms/SunwellPlateau/boss_kalecgos.cpp
-  EasternKingdoms/SunwellPlateau/boss_brutallus.cpp
-  EasternKingdoms/SunwellPlateau/sunwell_plateau.cpp
-  EasternKingdoms/SunwellPlateau/boss_felmyst.cpp
-  EasternKingdoms/stranglethorn_vale.cpp
-  EasternKingdoms/Deadmines/boss_glubtok.cpp
-  EasternKingdoms/Deadmines/boss_helix_gearbreaker.cpp
-  EasternKingdoms/Deadmines/boss_foereaper5000.cpp
-  EasternKingdoms/Deadmines/boss_admiral_ripsnarl.cpp
-  EasternKingdoms/Deadmines/boss_captain_cookie.cpp
-  EasternKingdoms/Deadmines/boss_vanessa_vancleef.cpp
-  EasternKingdoms/Deadmines/deadmines.h
-  EasternKingdoms/Deadmines/deadmines.cpp
-  EasternKingdoms/Deadmines/instance_deadmines.cpp
-  EasternKingdoms/duskwood.cpp
-  EasternKingdoms/ScarletMonastery/boss_azshir_the_sleepless.cpp
-  EasternKingdoms/ScarletMonastery/boss_mograine_and_whitemane.cpp
-  EasternKingdoms/ScarletMonastery/boss_bloodmage_thalnos.cpp
-  EasternKingdoms/ScarletMonastery/boss_interrogator_vishas.cpp
-  EasternKingdoms/ScarletMonastery/boss_headless_horseman.cpp
-  EasternKingdoms/ScarletMonastery/instance_scarlet_monastery.cpp
-  EasternKingdoms/ScarletMonastery/boss_houndmaster_loksey.cpp
-  EasternKingdoms/ScarletMonastery/scarlet_monastery.h
-  EasternKingdoms/ScarletMonastery/boss_high_inquisitor_fairbanks.cpp
-  EasternKingdoms/ScarletMonastery/boss_arcanist_doan.cpp
-  EasternKingdoms/ScarletMonastery/boss_herod.cpp
-  EasternKingdoms/ScarletMonastery/boss_scorn.cpp
-  EasternKingdoms/undercity.cpp
-  EasternKingdoms/silvermoon_city.cpp
-  EasternKingdoms/loch_modan.cpp
-  EasternKingdoms/ShadowfangKeep/boss_baron_ashbury.cpp
-  EasternKingdoms/ShadowfangKeep/boss_baron_silverlaine.cpp
-  EasternKingdoms/ShadowfangKeep/boss_commander_springvale.cpp
-  EasternKingdoms/ShadowfangKeep/boss_lord_valden.cpp
-  EasternKingdoms/ShadowfangKeep/boss_lord_godfrey.cpp
-  EasternKingdoms/ShadowfangKeep/shadowfang_keep.cpp
-  EasternKingdoms/ShadowfangKeep/instance_shadowfang_keep.cpp
-  EasternKingdoms/ShadowfangKeep/shadowfang_keep.h
-  EasternKingdoms/burning_steppes.cpp
-  EasternKingdoms/BlackwingLair/boss_chromaggus.cpp
-  EasternKingdoms/BlackwingLair/boss_razorgore.cpp
-  EasternKingdoms/BlackwingLair/boss_firemaw.cpp
-  EasternKingdoms/BlackwingLair/boss_broodlord_lashlayer.cpp
-  EasternKingdoms/BlackwingLair/boss_ebonroc.cpp
-  EasternKingdoms/BlackwingLair/instance_blackwing_lair.cpp
-  EasternKingdoms/BlackwingLair/blackwing_lair.h
-  EasternKingdoms/BlackwingLair/boss_vaelastrasz.cpp
-  EasternKingdoms/BlackwingLair/boss_nefarian.cpp
-  EasternKingdoms/BlackwingLair/boss_flamegor.cpp
-  EasternKingdoms/blasted_lands.cpp
-  EasternKingdoms/stormwind_city.cpp
-  EasternKingdoms/ZulAman/boss_halazzi.cpp
-  EasternKingdoms/ZulAman/boss_hexlord.cpp
-  EasternKingdoms/ZulAman/boss_daakara.cpp
-  EasternKingdoms/ZulAman/boss_akilzon.cpp
-  EasternKingdoms/ZulAman/instance_zulaman.cpp
-  EasternKingdoms/ZulAman/boss_janalai.cpp
-  EasternKingdoms/ZulAman/boss_nalorakk.cpp
-  EasternKingdoms/ZulAman/zulaman.cpp
-  EasternKingdoms/ZulAman/zulaman.h
-  EasternKingdoms/hinterlands.cpp
-  EasternKingdoms/western_plaguelands.cpp
-  EasternKingdoms/alterac_mountains.cpp
-  EasternKingdoms/westfall.cpp
-  EasternKingdoms/silverpine_forest.cpp
-  EasternKingdoms/deepholm.cpp
-  EasternKingdoms/Karazhan/instance_karazhan.cpp
-  EasternKingdoms/Karazhan/boss_nightbane.cpp
-  EasternKingdoms/Karazhan/karazhan.cpp
-  EasternKingdoms/Karazhan/boss_curator.cpp
-  EasternKingdoms/Karazhan/boss_shade_of_aran.cpp
-  EasternKingdoms/Karazhan/boss_netherspite.cpp
-  EasternKingdoms/Karazhan/boss_maiden_of_virtue.cpp
-  EasternKingdoms/Karazhan/boss_midnight.cpp
-  EasternKingdoms/Karazhan/boss_prince_malchezaar.cpp
-  EasternKingdoms/Karazhan/bosses_opera.cpp
-  EasternKingdoms/Karazhan/boss_moroes.cpp
-  EasternKingdoms/Karazhan/boss_terestian_illhoof.cpp
-  EasternKingdoms/Karazhan/karazhan.h
-  EasternKingdoms/ThroneOfTheTides/boss_lady_nazjar.cpp
-  EasternKingdoms/ThroneOfTheTides/boss_commander_ulthok.cpp
-  EasternKingdoms/ThroneOfTheTides/boss_mindebender_ghursha.cpp
-  EasternKingdoms/ThroneOfTheTides/boss_ozumat.cpp
-  EasternKingdoms/ThroneOfTheTides/instance_throne_of_the_tides.cpp
-  EasternKingdoms/ThroneOfTheTides/throne_of_the_tides.cpp
-  EasternKingdoms/ThroneOfTheTides/throne_of_the_tides.h
-  EasternKingdoms/BlackrockCaverns/boss_romogg_bonecrusher.cpp
-  EasternKingdoms/BlackrockCaverns/boss_corla_hearld_of_twilight.cpp
-  EasternKingdoms/BlackrockCaverns/boss_karsh_steelbender.cpp
-  EasternKingdoms/BlackrockCaverns/boss_beauty.cpp
-  EasternKingdoms/BlackrockCaverns/boss_ascendant_lord_obsidius.cpp
-  EasternKingdoms/BlackrockCaverns/instance_blackrock_caverns.cpp
-  EasternKingdoms/BlackrockCaverns/blackrock_caverns_teleport.cpp
-  EasternKingdoms/BlackrockCaverns/blackrock_caverns.h
-  EasternKingdoms/BaradinHold/boss_argaloth.cpp
-  EasternKingdoms/BaradinHold/boss_occuthar.cpp
-  EasternKingdoms/BaradinHold/boss_alizabal.cpp
-  EasternKingdoms/BaradinHold/instance_baradin_hold.cpp
-  EasternKingdoms/BaradinHold/baradin_hold.h
-  EasternKingdoms/TheStonecore/boss_corborus.cpp
-  EasternKingdoms/TheStonecore/boss_slabhide.cpp
-  EasternKingdoms/TheStonecore/boss_ozruk.cpp
-  EasternKingdoms/TheStonecore/boss_high_priestess_azil.cpp
-  EasternKingdoms/TheStonecore/instance_the_stonecore.cpp
-  EasternKingdoms/TheStonecore/the_stonecore.cpp
-  EasternKingdoms/TheStonecore/the_stonecore.h
-  EasternKingdoms/GrimBatol/boss_general_umbriss.cpp
-  EasternKingdoms/GrimBatol/boss_forgemaster_throngus.cpp
-  EasternKingdoms/GrimBatol/boss_drahga_shadowburner.cpp
-  EasternKingdoms/GrimBatol/boss_erudax.cpp
-  EasternKingdoms/GrimBatol/instance_grim_batol.cpp
-  EasternKingdoms/GrimBatol/grim_batol.cpp
-  EasternKingdoms/GrimBatol/grim_batol.h
-  EasternKingdoms/BastionOfTwilight/boss_halfus_wyrmbreaker.cpp
-  EasternKingdoms/BastionOfTwilight/boss_theralion_and_valiona.cpp
-  EasternKingdoms/BastionOfTwilight/boss_council_of_ascendents.cpp
-  EasternKingdoms/BastionOfTwilight/boss_chogall.cpp
-  EasternKingdoms/BastionOfTwilight/boss_sinestra.cpp
-  EasternKingdoms/BastionOfTwilight/instance_bastion_of_twilight.cpp
-  EasternKingdoms/BastionOfTwilight/bastion_of_twilight.cpp
-  EasternKingdoms/BastionOfTwilight/bastion_of_twilight.h
-  EasternKingdoms/BlackwingDescent/boss_magmaw.cpp
-  EasternKingdoms/BlackwingDescent/boss_omnotron_defence_system.cpp
-  EasternKingdoms/BlackwingDescent/boss_maloriak.cpp
-  EasternKingdoms/BlackwingDescent/boss_chimaeron.cpp
-  EasternKingdoms/BlackwingDescent/boss_atramedes.cpp
-  EasternKingdoms/BlackwingDescent/instance_blackwing_descent.cpp
-  EasternKingdoms/BlackwingDescent/blackwing_descent.cpp
-  EasternKingdoms/BlackwingDescent/blackwing_descent.h
+if( USE_EasternKingdomsPCH AND MSVC )
+  set(EasternKingdoms_STAT_SRCS
+    PrecompiledHeaders/EasternKingdomsPCH.cpp
+    PrecompiledHeaders/EasternKingdomsPCH.h
+  )
+endif()
+
+include_directories(
+  ${CMAKE_BINARY_DIR}
+  ${CMAKE_SOURCE_DIR}/dep/g3dlite/include
+  ${CMAKE_SOURCE_DIR}/dep/SFMT
+  ${CMAKE_SOURCE_DIR}/dep/zlib
+  ${CMAKE_SOURCE_DIR}/src/server/shared
+  ${CMAKE_SOURCE_DIR}/src/server/shared/Configuration
+  ${CMAKE_SOURCE_DIR}/src/server/shared/Cryptography
+  ${CMAKE_SOURCE_DIR}/src/server/shared/Cryptography/Authentication
+  ${CMAKE_SOURCE_DIR}/src/server/shared/Database
+  ${CMAKE_SOURCE_DIR}/src/server/shared/DataStores
+  ${CMAKE_SOURCE_DIR}/src/server/shared/Debugging
+  ${CMAKE_SOURCE_DIR}/src/server/shared/Dynamic/LinkedReference
+  ${CMAKE_SOURCE_DIR}/src/server/shared/Dynamic
+  ${CMAKE_SOURCE_DIR}/src/server/shared/Logging
+  ${CMAKE_SOURCE_DIR}/src/server/shared/Packets
+  ${CMAKE_SOURCE_DIR}/src/server/shared/Policies
+  ${CMAKE_SOURCE_DIR}/src/server/shared/Threading
+  ${CMAKE_SOURCE_DIR}/src/server/shared/Utilities
+  ${CMAKE_SOURCE_DIR}/src/server/collision
+  ${CMAKE_SOURCE_DIR}/src/server/collision/Management
+  ${CMAKE_SOURCE_DIR}/src/server/collision/Models
+  ${CMAKE_SOURCE_DIR}/src/server/shared
+  ${CMAKE_SOURCE_DIR}/src/server/shared/Database
+  ${CMAKE_SOURCE_DIR}/src/server/game/Accounts
+  ${CMAKE_SOURCE_DIR}/src/server/game/Achievements
+  ${CMAKE_SOURCE_DIR}/src/server/game/Addons
+  ${CMAKE_SOURCE_DIR}/src/server/game/AI
+  ${CMAKE_SOURCE_DIR}/src/server/game/AI/CoreAI
+  ${CMAKE_SOURCE_DIR}/src/server/game/AI/ScriptedAI
+  ${CMAKE_SOURCE_DIR}/src/server/game/AI/SmartScripts
+  ${CMAKE_SOURCE_DIR}/src/server/game/AuctionHouse
+  ${CMAKE_SOURCE_DIR}/src/server/game/Battlefield
+  ${CMAKE_SOURCE_DIR}/src/server/game/Battlefield/Zones
+  ${CMAKE_SOURCE_DIR}/src/server/game/Battlegrounds
+  ${CMAKE_SOURCE_DIR}/src/server/game/Battlegrounds/Zones
+  ${CMAKE_SOURCE_DIR}/src/server/game/Calendar
+  ${CMAKE_SOURCE_DIR}/src/server/game/Chat
+  ${CMAKE_SOURCE_DIR}/src/server/game/Chat/Channels
+  ${CMAKE_SOURCE_DIR}/src/server/game/Conditions
+  ${CMAKE_SOURCE_DIR}/src/server/shared/Configuration
+  ${CMAKE_SOURCE_DIR}/src/server/game/Combat
+  ${CMAKE_SOURCE_DIR}/src/server/game/DataStores
+  ${CMAKE_SOURCE_DIR}/src/server/game/DungeonFinding
+  ${CMAKE_SOURCE_DIR}/src/server/game/Entities/Corpse
+  ${CMAKE_SOURCE_DIR}/src/server/game/Entities/Creature
+  ${CMAKE_SOURCE_DIR}/src/server/game/Entities/DynamicObject
+  ${CMAKE_SOURCE_DIR}/src/server/game/Entities/Item
+  ${CMAKE_SOURCE_DIR}/src/server/game/Entities/Item/Container
+  ${CMAKE_SOURCE_DIR}/src/server/game/Entities/GameObject
+  ${CMAKE_SOURCE_DIR}/src/server/game/Entities/Object
+  ${CMAKE_SOURCE_DIR}/src/server/game/Entities/Object/Updates
+  ${CMAKE_SOURCE_DIR}/src/server/game/Entities/Pet
+  ${CMAKE_SOURCE_DIR}/src/server/game/Entities/Player
+  ${CMAKE_SOURCE_DIR}/src/server/game/Entities/Transport
+  ${CMAKE_SOURCE_DIR}/src/server/game/Entities/Unit
+  ${CMAKE_SOURCE_DIR}/src/server/game/Entities/Vehicle
+  ${CMAKE_SOURCE_DIR}/src/server/game/Events
+  ${CMAKE_SOURCE_DIR}/src/server/game/Globals
+  ${CMAKE_SOURCE_DIR}/src/server/game/Grids
+  ${CMAKE_SOURCE_DIR}/src/server/game/Grids/Cells
+  ${CMAKE_SOURCE_DIR}/src/server/game/Grids/Notifiers
+  ${CMAKE_SOURCE_DIR}/src/server/game/Groups
+  ${CMAKE_SOURCE_DIR}/src/server/game/Guilds
+  ${CMAKE_SOURCE_DIR}/src/server/game/Handlers
+  ${CMAKE_SOURCE_DIR}/src/server/game/Instances
+  ${CMAKE_SOURCE_DIR}/src/server/game/LookingForGroup
+  ${CMAKE_SOURCE_DIR}/src/server/game/Loot
+  ${CMAKE_SOURCE_DIR}/src/server/game/Mails
+  ${CMAKE_SOURCE_DIR}/src/server/game/Miscellaneous
+  ${CMAKE_SOURCE_DIR}/src/server/game/Maps
+  ${CMAKE_SOURCE_DIR}/src/server/game/Movement
+  ${CMAKE_SOURCE_DIR}/src/server/game/Movement/MovementGenerators
+  ${CMAKE_SOURCE_DIR}/src/server/game/Movement/Spline
+  ${CMAKE_SOURCE_DIR}/src/server/game/Movement/Waypoints
+  ${CMAKE_SOURCE_DIR}/src/server/game/Opcodes
+  ${CMAKE_SOURCE_DIR}/src/server/game/OutdoorPvP
+  ${CMAKE_SOURCE_DIR}/src/server/game/Pools
+  ${CMAKE_SOURCE_DIR}/src/server/game/PrecompiledHeaders
+  ${CMAKE_SOURCE_DIR}/src/server/game/Quests
+  ${CMAKE_SOURCE_DIR}/src/server/game/Reputation
+  ${CMAKE_SOURCE_DIR}/src/server/game/Scripting
+  ${CMAKE_SOURCE_DIR}/src/server/game/Server
+  ${CMAKE_SOURCE_DIR}/src/server/game/Server/Protocol
+  ${CMAKE_SOURCE_DIR}/src/server/game/Skills
+  ${CMAKE_SOURCE_DIR}/src/server/game/Spells
+  ${CMAKE_SOURCE_DIR}/src/server/game/Spells/Auras
+  ${CMAKE_SOURCE_DIR}/src/server/game/Texts
+  ${CMAKE_SOURCE_DIR}/src/server/game/Tickets
+  ${CMAKE_SOURCE_DIR}/src/server/game/Tools
+  ${CMAKE_SOURCE_DIR}/src/server/game/Warden
+  ${CMAKE_SOURCE_DIR}/src/server/game/Warden/Modules
+  ${CMAKE_SOURCE_DIR}/src/server/game/Weather
+  ${CMAKE_SOURCE_DIR}/src/server/game/World
+  ${CMAKE_SOURCE_DIR}/src/server/game/Entities/Totem
+  ${CMAKE_CURRENT_SOURCE_DIR}/PrecompiledHeaders
+  ${ACE_INCLUDE_DIR}
+  ${MYSQL_INCLUDE_DIR}
+  ${OPENSSL_INCLUDE_DIR}
+)
+
+set(EasternKingdoms_STAT_SRCS
+  ${EasternKingdoms_STAT_SRCS}
+  ghostlands.cpp
+  eversong_woods.cpp
+  AlteracValley/boss_galvangar.cpp
+  AlteracValley/boss_balinda.cpp
+  AlteracValley/boss_drekthar.cpp
+  AlteracValley/boss_vanndar.cpp
+  AlteracValley/alterac_valley.cpp
+  Scholomance/boss_the_ravenian.cpp
+  Scholomance/boss_instructor_malicia.cpp
+  Scholomance/boss_death_knight_darkreaver.cpp
+  Scholomance/boss_illucia_barov.cpp
+  Scholomance/scholomance.h
+  Scholomance/boss_vectus.cpp
+  Scholomance/boss_jandice_barov.cpp
+  Scholomance/boss_kormok.cpp
+  Scholomance/boss_lord_alexei_barov.cpp
+  Scholomance/boss_doctor_theolen_krastinov.cpp
+  Scholomance/boss_darkmaster_gandling.cpp
+  Scholomance/instance_scholomance.cpp
+  Scholomance/boss_lorekeeper_polkelt.cpp
+  Scholomance/boss_ras_frostwhisper.cpp
+  isle_of_queldanas.cpp
+  ZulGurub/boss_grilek.cpp
+  ZulGurub/boss_hazzarah.cpp
+  ZulGurub/boss_jindo_the_godbreaker.cpp
+  ZulGurub/boss_kilnara.cpp
+  ZulGurub/boss_mandokir.cpp
+  ZulGurub/boss_renataki.cpp
+  ZulGurub/boss_venoxis.cpp
+  ZulGurub/boss_wushoolay.cpp
+  ZulGurub/boss_zanzil.cpp
+  ZulGurub/instance_zulgurub.cpp
+  ZulGurub/zulgurub.cpp
+  ZulGurub/zulgurub.h
+  wetlands.cpp
+  arathi_highlands.cpp
+  Gnomeregan/instance_gnomeregan.cpp
+  Gnomeregan/gnomeregan.cpp
+  Gnomeregan/gnomeregan.h
+  redridge_mountains.cpp
+  BlackrockDepths/boss_high_interrogator_gerstahn.cpp
+  BlackrockDepths/boss_gorosh_the_dervish.cpp
+  BlackrockDepths/blackrock_depths.cpp
+  BlackrockDepths/boss_anubshiah.cpp
+  BlackrockDepths/boss_tomb_of_seven.cpp
+  BlackrockDepths/boss_general_angerforge.cpp
+  BlackrockDepths/boss_grizzle.cpp
+  BlackrockDepths/boss_ambassador_flamelash.cpp
+  BlackrockDepths/instance_blackrock_depths.cpp
+  BlackrockDepths/boss_moira_bronzebeard.cpp
+  BlackrockDepths/blackrock_depths.h
+  BlackrockDepths/boss_emperor_dagran_thaurissan.cpp
+  BlackrockDepths/boss_magmus.cpp
+  ironforge.cpp
+  ScarletEnclave/chapter2.cpp
+  ScarletEnclave/chapter5.cpp
+  ScarletEnclave/chapter1.cpp
+  ScarletEnclave/the_scarlet_enclave.cpp
+  eastern_plaguelands.cpp
+  MoltenCore/boss_gehennas.cpp
+  MoltenCore/boss_lucifron.cpp
+  MoltenCore/boss_golemagg.cpp
+  MoltenCore/boss_majordomo_executus.cpp
+  MoltenCore/boss_baron_geddon.cpp
+  MoltenCore/boss_ragnaros.cpp
+  MoltenCore/boss_garr.cpp
+  MoltenCore/molten_core.h
+  MoltenCore/instance_molten_core.cpp
+  MoltenCore/boss_sulfuron_harbinger.cpp
+  MoltenCore/boss_magmadar.cpp
+  MoltenCore/boss_shazzrah.cpp
+  Stratholme/boss_baroness_anastari.cpp
+  Stratholme/boss_nerubenkan.cpp
+  Stratholme/instance_stratholme.cpp
+  Stratholme/boss_dathrohan_balnazzar.cpp
+  Stratholme/boss_timmy_the_cruel.cpp
+  Stratholme/boss_baron_rivendare.cpp
+  Stratholme/boss_magistrate_barthilas.cpp
+  Stratholme/boss_order_of_silver_hand.cpp
+  Stratholme/boss_ramstein_the_gorger.cpp
+  Stratholme/boss_cannon_master_willey.cpp
+  Stratholme/boss_maleki_the_pallid.cpp
+  Stratholme/boss_postmaster_malown.cpp
+  Stratholme/stratholme.h
+  Stratholme/stratholme.cpp
+  tirisfal_glades.cpp
+  SunkenTemple/sunken_temple.cpp
+  SunkenTemple/sunken_temple.h
+  SunkenTemple/instance_sunken_temple.cpp
+  MagistersTerrace/boss_felblood_kaelthas.cpp
+  MagistersTerrace/magisters_terrace.h
+  MagistersTerrace/boss_priestess_delrissa.cpp
+  MagistersTerrace/instance_magisters_terrace.cpp
+  MagistersTerrace/boss_selin_fireheart.cpp
+  MagistersTerrace/boss_vexallus.cpp
+  MagistersTerrace/magisters_terrace.cpp
+  Uldaman/uldaman.cpp
+  Uldaman/boss_ironaya.cpp
+  Uldaman/uldaman.h
+  Uldaman/instance_uldaman.cpp
+  Uldaman/boss_archaedas.cpp
+  swamp_of_sorrows.cpp
+  BlackrockSpire/boss_pyroguard_emberseer.cpp
+  BlackrockSpire/boss_drakkisath.cpp
+  BlackrockSpire/boss_warmaster_voone.cpp
+  BlackrockSpire/boss_mother_smolderweb.cpp
+  BlackrockSpire/boss_quartermaster_zigris.cpp
+  BlackrockSpire/boss_halycon.cpp
+  BlackrockSpire/boss_overlord_wyrmthalak.cpp
+  BlackrockSpire/boss_shadow_hunter_voshgajin.cpp
+  BlackrockSpire/boss_gyth.cpp
+  BlackrockSpire/boss_rend_blackhand.cpp
+  BlackrockSpire/boss_highlord_omokk.cpp
+  BlackrockSpire/boss_the_beast.cpp
+  BlackrockSpire/blackrock_spire.h
+  BlackrockSpire/instance_blackrock_spire.cpp
+  SunwellPlateau/boss_eredar_twins.cpp
+  SunwellPlateau/boss_kiljaeden.cpp
+  SunwellPlateau/sunwell_plateau.h
+  SunwellPlateau/boss_muru.cpp
+  SunwellPlateau/instance_sunwell_plateau.cpp
+  SunwellPlateau/boss_kalecgos.cpp
+  SunwellPlateau/boss_brutallus.cpp
+  SunwellPlateau/sunwell_plateau.cpp
+  SunwellPlateau/boss_felmyst.cpp
+  stranglethorn_vale.cpp
+  Deadmines/boss_glubtok.cpp
+  Deadmines/boss_helix_gearbreaker.cpp
+  Deadmines/boss_foereaper5000.cpp
+  Deadmines/boss_admiral_ripsnarl.cpp
+  Deadmines/boss_captain_cookie.cpp
+  Deadmines/boss_vanessa_vancleef.cpp
+  Deadmines/deadmines.h
+  Deadmines/deadmines.cpp
+  Deadmines/instance_deadmines.cpp
+  duskwood.cpp
+  ScarletMonastery/boss_azshir_the_sleepless.cpp
+  ScarletMonastery/boss_mograine_and_whitemane.cpp
+  ScarletMonastery/boss_bloodmage_thalnos.cpp
+  ScarletMonastery/boss_interrogator_vishas.cpp
+  ScarletMonastery/boss_headless_horseman.cpp
+  ScarletMonastery/instance_scarlet_monastery.cpp
+  ScarletMonastery/boss_houndmaster_loksey.cpp
+  ScarletMonastery/scarlet_monastery.h
+  ScarletMonastery/boss_high_inquisitor_fairbanks.cpp
+  ScarletMonastery/boss_arcanist_doan.cpp
+  ScarletMonastery/boss_herod.cpp
+  ScarletMonastery/boss_scorn.cpp
+  undercity.cpp
+  silvermoon_city.cpp
+  loch_modan.cpp
+  ShadowfangKeep/boss_baron_ashbury.cpp
+  ShadowfangKeep/boss_baron_silverlaine.cpp
+  ShadowfangKeep/boss_commander_springvale.cpp
+  ShadowfangKeep/boss_lord_valden.cpp
+  ShadowfangKeep/boss_lord_godfrey.cpp
+  ShadowfangKeep/shadowfang_keep.cpp
+  ShadowfangKeep/instance_shadowfang_keep.cpp
+  ShadowfangKeep/shadowfang_keep.h
+  burning_steppes.cpp
+  BlackwingLair/boss_chromaggus.cpp
+  BlackwingLair/boss_razorgore.cpp
+  BlackwingLair/boss_firemaw.cpp
+  BlackwingLair/boss_broodlord_lashlayer.cpp
+  BlackwingLair/boss_ebonroc.cpp
+  BlackwingLair/instance_blackwing_lair.cpp
+  BlackwingLair/blackwing_lair.h
+  BlackwingLair/boss_vaelastrasz.cpp
+  BlackwingLair/boss_nefarian.cpp
+  BlackwingLair/boss_flamegor.cpp
+  blasted_lands.cpp
+  stormwind_city.cpp
+  ZulAman/boss_halazzi.cpp
+  ZulAman/boss_hexlord.cpp
+  ZulAman/boss_daakara.cpp
+  ZulAman/boss_akilzon.cpp
+  ZulAman/instance_zulaman.cpp
+  ZulAman/boss_janalai.cpp
+  ZulAman/boss_nalorakk.cpp
+  ZulAman/zulaman.cpp
+  ZulAman/zulaman.h
+  hinterlands.cpp
+  western_plaguelands.cpp
+  alterac_mountains.cpp
+  westfall.cpp
+  silverpine_forest.cpp
+  deepholm.cpp
+  Karazhan/instance_karazhan.cpp
+  Karazhan/boss_nightbane.cpp
+  Karazhan/karazhan.cpp
+  Karazhan/boss_curator.cpp
+  Karazhan/boss_shade_of_aran.cpp
+  Karazhan/boss_netherspite.cpp
+  Karazhan/boss_maiden_of_virtue.cpp
+  Karazhan/boss_midnight.cpp
+  Karazhan/boss_prince_malchezaar.cpp
+  Karazhan/bosses_opera.cpp
+  Karazhan/boss_moroes.cpp
+  Karazhan/boss_terestian_illhoof.cpp
+  Karazhan/karazhan.h
+  ThroneOfTheTides/boss_lady_nazjar.cpp
+  ThroneOfTheTides/boss_commander_ulthok.cpp
+  ThroneOfTheTides/boss_mindebender_ghursha.cpp
+  ThroneOfTheTides/boss_ozumat.cpp
+  ThroneOfTheTides/instance_throne_of_the_tides.cpp
+  ThroneOfTheTides/throne_of_the_tides.cpp
+  ThroneOfTheTides/throne_of_the_tides.h
+  BlackrockCaverns/boss_romogg_bonecrusher.cpp
+  BlackrockCaverns/boss_corla_hearld_of_twilight.cpp
+  BlackrockCaverns/boss_karsh_steelbender.cpp
+  BlackrockCaverns/boss_beauty.cpp
+  BlackrockCaverns/boss_ascendant_lord_obsidius.cpp
+  BlackrockCaverns/instance_blackrock_caverns.cpp
+  BlackrockCaverns/blackrock_caverns_teleport.cpp
+  BlackrockCaverns/blackrock_caverns.h
+  BaradinHold/boss_argaloth.cpp
+  BaradinHold/boss_occuthar.cpp
+  BaradinHold/boss_alizabal.cpp
+  BaradinHold/instance_baradin_hold.cpp
+  BaradinHold/baradin_hold.h
+  TheStonecore/boss_corborus.cpp
+  TheStonecore/boss_slabhide.cpp
+  TheStonecore/boss_ozruk.cpp
+  TheStonecore/boss_high_priestess_azil.cpp
+  TheStonecore/instance_the_stonecore.cpp
+  TheStonecore/the_stonecore.cpp
+  TheStonecore/the_stonecore.h
+  GrimBatol/boss_general_umbriss.cpp
+  GrimBatol/boss_forgemaster_throngus.cpp
+  GrimBatol/boss_drahga_shadowburner.cpp
+  GrimBatol/boss_erudax.cpp
+  GrimBatol/instance_grim_batol.cpp
+  GrimBatol/grim_batol.cpp
+  GrimBatol/grim_batol.h
+  BastionOfTwilight/boss_halfus_wyrmbreaker.cpp
+  BastionOfTwilight/boss_theralion_and_valiona.cpp
+  BastionOfTwilight/boss_council_of_ascendents.cpp
+  BastionOfTwilight/boss_chogall.cpp
+  BastionOfTwilight/boss_sinestra.cpp
+  BastionOfTwilight/instance_bastion_of_twilight.cpp
+  BastionOfTwilight/bastion_of_twilight.cpp
+  BastionOfTwilight/bastion_of_twilight.h
+  BlackwingDescent/boss_magmaw.cpp
+  BlackwingDescent/boss_omnotron_defence_system.cpp
+  BlackwingDescent/boss_maloriak.cpp
+  BlackwingDescent/boss_chimaeron.cpp
+  BlackwingDescent/boss_atramedes.cpp
+  BlackwingDescent/instance_blackwing_descent.cpp
+  BlackwingDescent/blackwing_descent.cpp
+  BlackwingDescent/blackwing_descent.h
 )
 
+add_library(EasternKingdoms STATIC ${EasternKingdoms_STAT_SRCS})
+
 message("  -> Prepared: Eastern Kingdoms")
+
+add_dependencies(EasternKingdoms revision.h)
+
+# Generate precompiled header
+if( USE_EasternKingdomsPCH )
+  if(CMAKE_COMPILER_IS_GNUCXX)
+    add_precompiled_header(EasternKingdoms ${CMAKE_CURRENT_SOURCE_DIR}/PrecompiledHeaders/EasternKingdomsPCH.h)
+  elseif(MSVC)
+    add_native_precompiled_header(EasternKingdoms ${CMAKE_CURRENT_SOURCE_DIR}/PrecompiledHeaders/EasternKingdomsPCH)
+  endif()
+endif()
\ No newline at end of file
diff --git a/src/server/scripts/EasternKingdoms/Deadmines/boss_admiral_ripsnarl.cpp b/src/server/scripts/EasternKingdoms/Deadmines/boss_admiral_ripsnarl.cpp
index 7e7773e..f6c7dfc 100644
--- a/src/server/scripts/EasternKingdoms/Deadmines/boss_admiral_ripsnarl.cpp
+++ b/src/server/scripts/EasternKingdoms/Deadmines/boss_admiral_ripsnarl.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "deadmines.h"
 
 enum ScriptTexts
diff --git a/src/server/scripts/EasternKingdoms/Deadmines/boss_captain_cookie.cpp b/src/server/scripts/EasternKingdoms/Deadmines/boss_captain_cookie.cpp
index 457182a..b27c546 100644
--- a/src/server/scripts/EasternKingdoms/Deadmines/boss_captain_cookie.cpp
+++ b/src/server/scripts/EasternKingdoms/Deadmines/boss_captain_cookie.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "deadmines.h"
 
 enum Spells
diff --git a/src/server/scripts/EasternKingdoms/Deadmines/boss_foereaper5000.cpp b/src/server/scripts/EasternKingdoms/Deadmines/boss_foereaper5000.cpp
index 01145a6..58b43c8 100644
--- a/src/server/scripts/EasternKingdoms/Deadmines/boss_foereaper5000.cpp
+++ b/src/server/scripts/EasternKingdoms/Deadmines/boss_foereaper5000.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "deadmines.h"
 
 
diff --git a/src/server/scripts/EasternKingdoms/Deadmines/boss_glubtok.cpp b/src/server/scripts/EasternKingdoms/Deadmines/boss_glubtok.cpp
index 72c2307..a5a715a 100644
--- a/src/server/scripts/EasternKingdoms/Deadmines/boss_glubtok.cpp
+++ b/src/server/scripts/EasternKingdoms/Deadmines/boss_glubtok.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "deadmines.h"
 
 //todo: реализовать flame wall
diff --git a/src/server/scripts/EasternKingdoms/Deadmines/boss_helix_gearbreaker.cpp b/src/server/scripts/EasternKingdoms/Deadmines/boss_helix_gearbreaker.cpp
index 9216cd1..924122f 100644
--- a/src/server/scripts/EasternKingdoms/Deadmines/boss_helix_gearbreaker.cpp
+++ b/src/server/scripts/EasternKingdoms/Deadmines/boss_helix_gearbreaker.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "deadmines.h"
 #include "Vehicle.h"
 
diff --git a/src/server/scripts/EasternKingdoms/Deadmines/boss_vanessa_vancleef.cpp b/src/server/scripts/EasternKingdoms/Deadmines/boss_vanessa_vancleef.cpp
index 3a5d500..efc4364 100644
--- a/src/server/scripts/EasternKingdoms/Deadmines/boss_vanessa_vancleef.cpp
+++ b/src/server/scripts/EasternKingdoms/Deadmines/boss_vanessa_vancleef.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "deadmines.h"
 
 enum ScriptedTexts
diff --git a/src/server/scripts/EasternKingdoms/Deadmines/deadmines.cpp b/src/server/scripts/EasternKingdoms/Deadmines/deadmines.cpp
index d9af530..4b4220e 100644
--- a/src/server/scripts/EasternKingdoms/Deadmines/deadmines.cpp
+++ b/src/server/scripts/EasternKingdoms/Deadmines/deadmines.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "ScriptedGossip.h"
 #include "ScriptMgr.h"
 #include "InstanceScript.h"
diff --git a/src/server/scripts/EasternKingdoms/Deadmines/instance_deadmines.cpp b/src/server/scripts/EasternKingdoms/Deadmines/instance_deadmines.cpp
index 1ddf4e3..103fcc3 100644
--- a/src/server/scripts/EasternKingdoms/Deadmines/instance_deadmines.cpp
+++ b/src/server/scripts/EasternKingdoms/Deadmines/instance_deadmines.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "deadmines.h"
 
 #define MAX_ENCOUNTER 6
diff --git a/src/server/scripts/EasternKingdoms/GrimBatol/boss_drahga_shadowburner.cpp b/src/server/scripts/EasternKingdoms/GrimBatol/boss_drahga_shadowburner.cpp
index 3f4e59f..7f30f15 100644
--- a/src/server/scripts/EasternKingdoms/GrimBatol/boss_drahga_shadowburner.cpp
+++ b/src/server/scripts/EasternKingdoms/GrimBatol/boss_drahga_shadowburner.cpp
@@ -1,6 +1,6 @@
 #include "grim_batol.h"
 #include "Spell.h"
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "Vehicle.h"
 
 
diff --git a/src/server/scripts/EasternKingdoms/GrimBatol/boss_erudax.cpp b/src/server/scripts/EasternKingdoms/GrimBatol/boss_erudax.cpp
index 3a7ccda..83dbf73 100644
--- a/src/server/scripts/EasternKingdoms/GrimBatol/boss_erudax.cpp
+++ b/src/server/scripts/EasternKingdoms/GrimBatol/boss_erudax.cpp
@@ -1,5 +1,5 @@
 #include "grim_batol.h"
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 
 
 //todo: щит ночных кошмаров, аое, евент с адами
diff --git a/src/server/scripts/EasternKingdoms/GrimBatol/boss_forgemaster_throngus.cpp b/src/server/scripts/EasternKingdoms/GrimBatol/boss_forgemaster_throngus.cpp
index 43474eb..3c54360 100644
--- a/src/server/scripts/EasternKingdoms/GrimBatol/boss_forgemaster_throngus.cpp
+++ b/src/server/scripts/EasternKingdoms/GrimBatol/boss_forgemaster_throngus.cpp
@@ -1,5 +1,5 @@
 #include "grim_batol.h"
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 
 
 //todo: реализовать адов и mighty stomp
diff --git a/src/server/scripts/EasternKingdoms/GrimBatol/boss_general_umbriss.cpp b/src/server/scripts/EasternKingdoms/GrimBatol/boss_general_umbriss.cpp
index e6ddc5f..59d985c 100644
--- a/src/server/scripts/EasternKingdoms/GrimBatol/boss_general_umbriss.cpp
+++ b/src/server/scripts/EasternKingdoms/GrimBatol/boss_general_umbriss.cpp
@@ -1,5 +1,5 @@
 #include "grim_batol.h"
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 
 enum ScriptTexts
 {
diff --git a/src/server/scripts/EasternKingdoms/GrimBatol/grim_batol.cpp b/src/server/scripts/EasternKingdoms/GrimBatol/grim_batol.cpp
index a6f291f..42ad8fb 100644
--- a/src/server/scripts/EasternKingdoms/GrimBatol/grim_batol.cpp
+++ b/src/server/scripts/EasternKingdoms/GrimBatol/grim_batol.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "grim_batol.h"
     
 enum Creatures
diff --git a/src/server/scripts/EasternKingdoms/GrimBatol/instance_grim_batol.cpp b/src/server/scripts/EasternKingdoms/GrimBatol/instance_grim_batol.cpp
index 91b9152..1f6a556 100644
--- a/src/server/scripts/EasternKingdoms/GrimBatol/instance_grim_batol.cpp
+++ b/src/server/scripts/EasternKingdoms/GrimBatol/instance_grim_batol.cpp
@@ -1,5 +1,5 @@
 #include "grim_batol.h"
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 
 class instance_grim_batol : public InstanceMapScript
 {
diff --git a/src/server/scripts/EasternKingdoms/PrecompiledHeaders/EasternKingdomsPCH.cpp b/src/server/scripts/EasternKingdoms/PrecompiledHeaders/EasternKingdomsPCH.cpp
new file mode 100644
index 0000000..3350280
--- /dev/null
+++ b/src/server/scripts/EasternKingdoms/PrecompiledHeaders/EasternKingdomsPCH.cpp
@@ -0,0 +1,6 @@
+/* Copyright (C) 2006 - 2009 ScriptDev2 <https://scriptdev2.svn.sourceforge.net/>
+* This program is free software licensed under GPL version 2
+* Please see the included DOCS/LICENSE.TXT for more information */
+
+#include "EasternKingdomsPCH.h"
+
diff --git a/src/server/scripts/EasternKingdoms/PrecompiledHeaders/EasternKingdomsPCH.h b/src/server/scripts/EasternKingdoms/PrecompiledHeaders/EasternKingdomsPCH.h
new file mode 100644
index 0000000..823054e
--- /dev/null
+++ b/src/server/scripts/EasternKingdoms/PrecompiledHeaders/EasternKingdomsPCH.h
@@ -0,0 +1,34 @@
+/* Copyright (C) 2006 - 2009 ScriptDev2 <https://scriptdev2.svn.sourceforge.net/>
+* This program is free software licensed under GPL version 2
+* Please see the included DOCS/LICENSE.TXT for more information */
+
+#ifndef SC_PRECOMPILED_H
+#define SC_PRECOMPILED_H
+
+#include "ScriptMgr.h"
+#include "Cell.h"
+#include "CellImpl.h"
+#include "GameEventMgr.h"
+#include "GridNotifiers.h"
+#include "GridNotifiersImpl.h"
+#include "Unit.h"
+#include "GameObject.h"
+#include "ScriptedCreature.h"
+#include "ScriptedGossip.h"
+#include "InstanceScript.h"
+#include "CombatAI.h"
+#include "PassiveAI.h"
+#include "Chat.h"
+#include "DBCStructure.h"
+#include "DBCStores.h"
+#include "ObjectMgr.h"
+#include "SpellScript.h"
+#include "SpellAuraEffects.h"
+#include "Vehicle.h"
+
+#ifdef _WIN32
+#include <windows.h>
+#endif
+
+#endif
+
diff --git a/src/server/scripts/EasternKingdoms/ShadowfangKeep/boss_baron_ashbury.cpp b/src/server/scripts/EasternKingdoms/ShadowfangKeep/boss_baron_ashbury.cpp
index 809a278..ecc218e 100644
--- a/src/server/scripts/EasternKingdoms/ShadowfangKeep/boss_baron_ashbury.cpp
+++ b/src/server/scripts/EasternKingdoms/ShadowfangKeep/boss_baron_ashbury.cpp
@@ -1,4 +1,4 @@
-#include"ScriptPCH.h"
+#include"EasternKingdomsPCH.h"
 #include"Spell.h"
 #include"shadowfang_keep.h"
 
diff --git a/src/server/scripts/EasternKingdoms/ShadowfangKeep/boss_baron_silverlaine.cpp b/src/server/scripts/EasternKingdoms/ShadowfangKeep/boss_baron_silverlaine.cpp
index 1db855c..9ee35cb 100644
--- a/src/server/scripts/EasternKingdoms/ShadowfangKeep/boss_baron_silverlaine.cpp
+++ b/src/server/scripts/EasternKingdoms/ShadowfangKeep/boss_baron_silverlaine.cpp
@@ -1,4 +1,4 @@
-#include"ScriptPCH.h"
+#include"EasternKingdomsPCH.h"
 #include"shadowfang_keep.h"
 
 enum ScriptTexts
diff --git a/src/server/scripts/EasternKingdoms/ShadowfangKeep/boss_commander_springvale.cpp b/src/server/scripts/EasternKingdoms/ShadowfangKeep/boss_commander_springvale.cpp
index 3d5c6d5..b1f9f92 100644
--- a/src/server/scripts/EasternKingdoms/ShadowfangKeep/boss_commander_springvale.cpp
+++ b/src/server/scripts/EasternKingdoms/ShadowfangKeep/boss_commander_springvale.cpp
@@ -1,4 +1,4 @@
-#include"ScriptPCH.h"
+#include"EasternKingdomsPCH.h"
 #include"WorldPacket.h"
 #include"shadowfang_keep.h"
 #include"ScriptMgr.h"
diff --git a/src/server/scripts/EasternKingdoms/ShadowfangKeep/boss_lord_godfrey.cpp b/src/server/scripts/EasternKingdoms/ShadowfangKeep/boss_lord_godfrey.cpp
index a8f147e..7ba388b 100644
--- a/src/server/scripts/EasternKingdoms/ShadowfangKeep/boss_lord_godfrey.cpp
+++ b/src/server/scripts/EasternKingdoms/ShadowfangKeep/boss_lord_godfrey.cpp
@@ -1,4 +1,4 @@
-#include"ScriptPCH.h"
+#include"EasternKingdomsPCH.h"
 #include"Spell.h"
 #include"shadowfang_keep.h"
 
diff --git a/src/server/scripts/EasternKingdoms/ShadowfangKeep/boss_lord_valden.cpp b/src/server/scripts/EasternKingdoms/ShadowfangKeep/boss_lord_valden.cpp
index 5c1c0d2..bf799fe 100644
--- a/src/server/scripts/EasternKingdoms/ShadowfangKeep/boss_lord_valden.cpp
+++ b/src/server/scripts/EasternKingdoms/ShadowfangKeep/boss_lord_valden.cpp
@@ -1,4 +1,4 @@
-#include"ScriptPCH.h"
+#include"EasternKingdomsPCH.h"
 #include"shadowfang_keep.h"
 
 enum ScriptTexts
diff --git a/src/server/scripts/EasternKingdoms/ShadowfangKeep/instance_shadowfang_keep.cpp b/src/server/scripts/EasternKingdoms/ShadowfangKeep/instance_shadowfang_keep.cpp
index 9b8928d..6d66f1e 100644
--- a/src/server/scripts/EasternKingdoms/ShadowfangKeep/instance_shadowfang_keep.cpp
+++ b/src/server/scripts/EasternKingdoms/ShadowfangKeep/instance_shadowfang_keep.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "shadowfang_keep.h"
 
 DoorData const doorData[] =
diff --git a/src/server/scripts/EasternKingdoms/ShadowfangKeep/shadowfang_keep.cpp b/src/server/scripts/EasternKingdoms/ShadowfangKeep/shadowfang_keep.cpp
index 93d5c27..16e7f1a 100644
--- a/src/server/scripts/EasternKingdoms/ShadowfangKeep/shadowfang_keep.cpp
+++ b/src/server/scripts/EasternKingdoms/ShadowfangKeep/shadowfang_keep.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "LFGMgr.h"
 #include "Group.h"
 #include "shadowfang_keep.h"
diff --git a/src/server/scripts/EasternKingdoms/TheStonecore/boss_corborus.cpp b/src/server/scripts/EasternKingdoms/TheStonecore/boss_corborus.cpp
index cb3c489..6ce941c 100644
--- a/src/server/scripts/EasternKingdoms/TheStonecore/boss_corborus.cpp
+++ b/src/server/scripts/EasternKingdoms/TheStonecore/boss_corborus.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "the_stonecore.h"
 
 enum Spells
diff --git a/src/server/scripts/EasternKingdoms/TheStonecore/boss_high_priestess_azil.cpp b/src/server/scripts/EasternKingdoms/TheStonecore/boss_high_priestess_azil.cpp
index d1369da..6733741 100644
--- a/src/server/scripts/EasternKingdoms/TheStonecore/boss_high_priestess_azil.cpp
+++ b/src/server/scripts/EasternKingdoms/TheStonecore/boss_high_priestess_azil.cpp
@@ -1,5 +1,5 @@
 
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "the_stonecore.h"
 
 enum ScriptTexts
diff --git a/src/server/scripts/EasternKingdoms/TheStonecore/boss_ozruk.cpp b/src/server/scripts/EasternKingdoms/TheStonecore/boss_ozruk.cpp
index ee54eba..63fe51c 100644
--- a/src/server/scripts/EasternKingdoms/TheStonecore/boss_ozruk.cpp
+++ b/src/server/scripts/EasternKingdoms/TheStonecore/boss_ozruk.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "the_stonecore.h"
 
 enum ScriptTexts
diff --git a/src/server/scripts/EasternKingdoms/TheStonecore/boss_slabhide.cpp b/src/server/scripts/EasternKingdoms/TheStonecore/boss_slabhide.cpp
index 6c18b42..712b979 100644
--- a/src/server/scripts/EasternKingdoms/TheStonecore/boss_slabhide.cpp
+++ b/src/server/scripts/EasternKingdoms/TheStonecore/boss_slabhide.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "the_stonecore.h"
 
 enum Spells
diff --git a/src/server/scripts/EasternKingdoms/TheStonecore/instance_the_stonecore.cpp b/src/server/scripts/EasternKingdoms/TheStonecore/instance_the_stonecore.cpp
index aea3281..7dedc81 100644
--- a/src/server/scripts/EasternKingdoms/TheStonecore/instance_the_stonecore.cpp
+++ b/src/server/scripts/EasternKingdoms/TheStonecore/instance_the_stonecore.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "the_stonecore.h"
 
 class instance_the_stonecore : public InstanceMapScript
diff --git a/src/server/scripts/EasternKingdoms/TheStonecore/the_stonecore.cpp b/src/server/scripts/EasternKingdoms/TheStonecore/the_stonecore.cpp
index f5f06cd..9beef6d 100644
--- a/src/server/scripts/EasternKingdoms/TheStonecore/the_stonecore.cpp
+++ b/src/server/scripts/EasternKingdoms/TheStonecore/the_stonecore.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "the_stonecore.h"
 
 enum Spells
diff --git a/src/server/scripts/EasternKingdoms/ThroneOfTheTides/boss_commander_ulthok.cpp b/src/server/scripts/EasternKingdoms/ThroneOfTheTides/boss_commander_ulthok.cpp
index 03529c9..2e4dd93 100644
--- a/src/server/scripts/EasternKingdoms/ThroneOfTheTides/boss_commander_ulthok.cpp
+++ b/src/server/scripts/EasternKingdoms/ThroneOfTheTides/boss_commander_ulthok.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "throne_of_the_tides.h"
 
 enum Spells
diff --git a/src/server/scripts/EasternKingdoms/ThroneOfTheTides/boss_lady_nazjar.cpp b/src/server/scripts/EasternKingdoms/ThroneOfTheTides/boss_lady_nazjar.cpp
index a9c60fc..ec38ed9 100644
--- a/src/server/scripts/EasternKingdoms/ThroneOfTheTides/boss_lady_nazjar.cpp
+++ b/src/server/scripts/EasternKingdoms/ThroneOfTheTides/boss_lady_nazjar.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "Spell.h"
 #include "throne_of_the_tides.h"
 
diff --git a/src/server/scripts/EasternKingdoms/ThroneOfTheTides/boss_mindebender_ghursha.cpp b/src/server/scripts/EasternKingdoms/ThroneOfTheTides/boss_mindebender_ghursha.cpp
index 3b5b943..d698616 100644
--- a/src/server/scripts/EasternKingdoms/ThroneOfTheTides/boss_mindebender_ghursha.cpp
+++ b/src/server/scripts/EasternKingdoms/ThroneOfTheTides/boss_mindebender_ghursha.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "Spell.h"
 #include "throne_of_the_tides.h"
 
diff --git a/src/server/scripts/EasternKingdoms/ThroneOfTheTides/boss_ozumat.cpp b/src/server/scripts/EasternKingdoms/ThroneOfTheTides/boss_ozumat.cpp
index 031fcff..d0243f4 100644
--- a/src/server/scripts/EasternKingdoms/ThroneOfTheTides/boss_ozumat.cpp
+++ b/src/server/scripts/EasternKingdoms/ThroneOfTheTides/boss_ozumat.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "throne_of_the_tides.h"
 #include "Group.h"
 
diff --git a/src/server/scripts/EasternKingdoms/ThroneOfTheTides/instance_throne_of_the_tides.cpp b/src/server/scripts/EasternKingdoms/ThroneOfTheTides/instance_throne_of_the_tides.cpp
index a78e651..bf8c2ac 100644
--- a/src/server/scripts/EasternKingdoms/ThroneOfTheTides/instance_throne_of_the_tides.cpp
+++ b/src/server/scripts/EasternKingdoms/ThroneOfTheTides/instance_throne_of_the_tides.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "throne_of_the_tides.h"
 
 #define MAX_ENCOUNTER 4
diff --git a/src/server/scripts/EasternKingdoms/ThroneOfTheTides/throne_of_the_tides.cpp b/src/server/scripts/EasternKingdoms/ThroneOfTheTides/throne_of_the_tides.cpp
index 751ae20..41dd5da 100644
--- a/src/server/scripts/EasternKingdoms/ThroneOfTheTides/throne_of_the_tides.cpp
+++ b/src/server/scripts/EasternKingdoms/ThroneOfTheTides/throne_of_the_tides.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "throne_of_the_tides.h"
 
 enum ScriptTexts
diff --git a/src/server/scripts/EasternKingdoms/ZulAman/boss_akilzon.cpp b/src/server/scripts/EasternKingdoms/ZulAman/boss_akilzon.cpp
index dbcd203..26a0d2d 100644
--- a/src/server/scripts/EasternKingdoms/ZulAman/boss_akilzon.cpp
+++ b/src/server/scripts/EasternKingdoms/ZulAman/boss_akilzon.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "zulaman.h"
 
 #define SE_LOC_X_MAX 400
diff --git a/src/server/scripts/EasternKingdoms/ZulAman/boss_daakara.cpp b/src/server/scripts/EasternKingdoms/ZulAman/boss_daakara.cpp
index 6b3ff5c..4d02a71 100644
--- a/src/server/scripts/EasternKingdoms/ZulAman/boss_daakara.cpp
+++ b/src/server/scripts/EasternKingdoms/ZulAman/boss_daakara.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "zulaman.h"
 
 enum Scripttexts
diff --git a/src/server/scripts/EasternKingdoms/ZulAman/boss_halazzi.cpp b/src/server/scripts/EasternKingdoms/ZulAman/boss_halazzi.cpp
index 9176daa..ba1dec1 100644
--- a/src/server/scripts/EasternKingdoms/ZulAman/boss_halazzi.cpp
+++ b/src/server/scripts/EasternKingdoms/ZulAman/boss_halazzi.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "zulaman.h"
 
 enum ScriptTexts
diff --git a/src/server/scripts/EasternKingdoms/ZulAman/boss_hexlord.cpp b/src/server/scripts/EasternKingdoms/ZulAman/boss_hexlord.cpp
index 21dcef5..b4d5dc0 100644
--- a/src/server/scripts/EasternKingdoms/ZulAman/boss_hexlord.cpp
+++ b/src/server/scripts/EasternKingdoms/ZulAman/boss_hexlord.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "zulaman.h"
 
 enum ScriptTexts
diff --git a/src/server/scripts/EasternKingdoms/ZulAman/boss_janalai.cpp b/src/server/scripts/EasternKingdoms/ZulAman/boss_janalai.cpp
index 0c1d422..4316cec 100644
--- a/src/server/scripts/EasternKingdoms/ZulAman/boss_janalai.cpp
+++ b/src/server/scripts/EasternKingdoms/ZulAman/boss_janalai.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "zulaman.h"
 
 enum ScriptTexts
diff --git a/src/server/scripts/EasternKingdoms/ZulAman/boss_nalorakk.cpp b/src/server/scripts/EasternKingdoms/ZulAman/boss_nalorakk.cpp
index 1871c19..97ed016 100644
--- a/src/server/scripts/EasternKingdoms/ZulAman/boss_nalorakk.cpp
+++ b/src/server/scripts/EasternKingdoms/ZulAman/boss_nalorakk.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "zulaman.h"
 
 enum ScriptTexts
diff --git a/src/server/scripts/EasternKingdoms/ZulAman/instance_zulaman.cpp b/src/server/scripts/EasternKingdoms/ZulAman/instance_zulaman.cpp
index 60367d6..da901b4 100644
--- a/src/server/scripts/EasternKingdoms/ZulAman/instance_zulaman.cpp
+++ b/src/server/scripts/EasternKingdoms/ZulAman/instance_zulaman.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "zulaman.h"
 
 #define MAX_ENCOUNTER 6
diff --git a/src/server/scripts/EasternKingdoms/ZulAman/zulaman.cpp b/src/server/scripts/EasternKingdoms/ZulAman/zulaman.cpp
index c7e0601..936883c 100644
--- a/src/server/scripts/EasternKingdoms/ZulAman/zulaman.cpp
+++ b/src/server/scripts/EasternKingdoms/ZulAman/zulaman.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "zulaman.h"
 
 enum Spells
diff --git a/src/server/scripts/EasternKingdoms/ZulGurub/boss_grilek.cpp b/src/server/scripts/EasternKingdoms/ZulGurub/boss_grilek.cpp
index 18a806b..3f599aa 100644
--- a/src/server/scripts/EasternKingdoms/ZulGurub/boss_grilek.cpp
+++ b/src/server/scripts/EasternKingdoms/ZulGurub/boss_grilek.cpp
@@ -1,6 +1,4 @@
-#include "ObjectMgr.h"
-#include "ScriptMgr.h"
-#include "ScriptedCreature.h"
+#include "EasternKingdomsPCH.h"
 #include "zulgurub.h"
 
 enum Yells
diff --git a/src/server/scripts/EasternKingdoms/ZulGurub/boss_hazzarah.cpp b/src/server/scripts/EasternKingdoms/ZulGurub/boss_hazzarah.cpp
index 1ef95e4..b590ec8 100644
--- a/src/server/scripts/EasternKingdoms/ZulGurub/boss_hazzarah.cpp
+++ b/src/server/scripts/EasternKingdoms/ZulGurub/boss_hazzarah.cpp
@@ -1,6 +1,4 @@
-#include "ObjectMgr.h"
-#include "ScriptMgr.h"
-#include "ScriptedCreature.h"
+#include "EasternKingdomsPCH.h"
 #include "zulgurub.h"
 
 enum Yells
diff --git a/src/server/scripts/EasternKingdoms/ZulGurub/boss_jindo_the_godbreaker.cpp b/src/server/scripts/EasternKingdoms/ZulGurub/boss_jindo_the_godbreaker.cpp
index 58d9bb9..bfeb108 100644
--- a/src/server/scripts/EasternKingdoms/ZulGurub/boss_jindo_the_godbreaker.cpp
+++ b/src/server/scripts/EasternKingdoms/ZulGurub/boss_jindo_the_godbreaker.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "zulgurub.h"
 
 enum JindoScriptTexts
diff --git a/src/server/scripts/EasternKingdoms/ZulGurub/boss_kilnara.cpp b/src/server/scripts/EasternKingdoms/ZulGurub/boss_kilnara.cpp
index 4405ac7..738373b 100644
--- a/src/server/scripts/EasternKingdoms/ZulGurub/boss_kilnara.cpp
+++ b/src/server/scripts/EasternKingdoms/ZulGurub/boss_kilnara.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "Spell.h"
 #include "zulgurub.h"
 
diff --git a/src/server/scripts/EasternKingdoms/ZulGurub/boss_renataki.cpp b/src/server/scripts/EasternKingdoms/ZulGurub/boss_renataki.cpp
index 2e35106..835acc5 100644
--- a/src/server/scripts/EasternKingdoms/ZulGurub/boss_renataki.cpp
+++ b/src/server/scripts/EasternKingdoms/ZulGurub/boss_renataki.cpp
@@ -1,6 +1,4 @@
-#include "ObjectMgr.h"
-#include "ScriptMgr.h"
-#include "ScriptedCreature.h"
+#include "EasternKingdomsPCH.h"
 #include "zulgurub.h"
 
 enum Yells
diff --git a/src/server/scripts/EasternKingdoms/ZulGurub/boss_venoxis.cpp b/src/server/scripts/EasternKingdoms/ZulGurub/boss_venoxis.cpp
index 7c25537..c86f2ac 100644
--- a/src/server/scripts/EasternKingdoms/ZulGurub/boss_venoxis.cpp
+++ b/src/server/scripts/EasternKingdoms/ZulGurub/boss_venoxis.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 //#include "GridNotifiers.h"
 #include "zulgurub.h"
 
diff --git a/src/server/scripts/EasternKingdoms/ZulGurub/boss_wushoolay.cpp b/src/server/scripts/EasternKingdoms/ZulGurub/boss_wushoolay.cpp
index f223393..1018d11 100644
--- a/src/server/scripts/EasternKingdoms/ZulGurub/boss_wushoolay.cpp
+++ b/src/server/scripts/EasternKingdoms/ZulGurub/boss_wushoolay.cpp
@@ -1,6 +1,4 @@
-#include "ObjectMgr.h"
-#include "ScriptMgr.h"
-#include "ScriptedCreature.h"
+#include "EasternKingdomsPCH.h"
 #include "zulgurub.h"
 
 enum Yells
diff --git a/src/server/scripts/EasternKingdoms/ZulGurub/boss_zanzil.cpp b/src/server/scripts/EasternKingdoms/ZulGurub/boss_zanzil.cpp
index e36d156..e641c2f 100644
--- a/src/server/scripts/EasternKingdoms/ZulGurub/boss_zanzil.cpp
+++ b/src/server/scripts/EasternKingdoms/ZulGurub/boss_zanzil.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "zulgurub.h"
 
 enum ScriptTexts
diff --git a/src/server/scripts/EasternKingdoms/ZulGurub/instance_zulgurub.cpp b/src/server/scripts/EasternKingdoms/ZulGurub/instance_zulgurub.cpp
index c93f03e..f34cd75 100644
--- a/src/server/scripts/EasternKingdoms/ZulGurub/instance_zulgurub.cpp
+++ b/src/server/scripts/EasternKingdoms/ZulGurub/instance_zulgurub.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 #include "zulgurub.h"
 
 static const DoorData doordata[] = 
diff --git a/src/server/scripts/EasternKingdoms/ZulGurub/zulgurub.cpp b/src/server/scripts/EasternKingdoms/ZulGurub/zulgurub.cpp
index b7fb9c7..c8b3a5c 100644
--- a/src/server/scripts/EasternKingdoms/ZulGurub/zulgurub.cpp
+++ b/src/server/scripts/EasternKingdoms/ZulGurub/zulgurub.cpp
@@ -1,4 +1,4 @@
-#include"ScriptPCH.h"
+#include"EasternKingdomsPCH.h"
 #include "zulgurub.h"
 
 enum Spells
diff --git a/src/server/scripts/EasternKingdoms/deepholm.cpp b/src/server/scripts/EasternKingdoms/deepholm.cpp
index 43fc887..c63d20c 100644
--- a/src/server/scripts/EasternKingdoms/deepholm.cpp
+++ b/src/server/scripts/EasternKingdoms/deepholm.cpp
@@ -1,4 +1,4 @@
-#include "ScriptPCH.h"
+#include "EasternKingdomsPCH.h"
 
 enum CreatureIds
 {
diff --git a/src/server/worldserver/CMakeLists.txt b/src/server/worldserver/CMakeLists.txt
index 364e8e7..ee47d7a 100644
--- a/src/server/worldserver/CMakeLists.txt
+++ b/src/server/worldserver/CMakeLists.txt
@@ -162,6 +162,7 @@ target_link_libraries(worldserver
   game
   shared
   scripts
+  EasternKingdoms
   collision
   g3dlib
   gsoap
